feat(configuration): Merging in prior release changes.

cgaber2045 opened the pull request.

cgaber2045 committed:
feat(configuration): Update private.py environment configuration.

cgaber2045 committed:
feat(configuration): Redirecting from dashboard page to resume course with flag.

cgaber2045 committed:
feat(configuration): ALWAYS_REDIRECT_HOMEPAGE_TO_LOGIN_FOR_UNAUTHENTICATED_USER Site Configuration.

ztraboo commented on lms/djangoapps/branding/views.py:
```diff
@@ -34,6 +35,14 @@ def index(request):
     Redirects to main page -- info page if user authenticated, or marketing if not
     """
+    # This is to redirect Microsites when a homepage is not necessarily needed.
+    if not request.user.is_authenticated():
```
@cgaber2045 I realize you pulled in this code from a prior release, however, please make sure to test it when possible.

When I tested this it produced this error when the user was logged out of the Open edX LMS and I traversed to the http://localhost:18000/ page.
> (Pdb) request.user.is_authenticated()
*** TypeError: 'bool' object is not callable

To resolve this remove the `()` on the `is_authenticated` property. I'll go ahead and make this change then merge.

cc: @becdavid

ztraboo committed:
fix(configuration): Remove the unneeded `()` for property.

When I tested this it produced this error when the user was logged out of the Open edX LMS and I traversed to the http://localhost:18000/ page.

> (Pdb) request.user.is_authenticated()
*** TypeError: 'bool' object is not callable

To resolve this remove the () on the is_authenticated property.

https://github.com/CUCWD/edx-platform/pull/156/files#r841815053
